src/main/java/com/cloudant/client/api/CloudantClient.java

```diff
@@ -236,7 +237,30 @@ public CloudantClient(String account, String authCookie, ConnectOptions connectO
         doInit(h.get("scheme"),
                 h.get("hostname"),
                 new Integer(h.get("port")).intValue(),
-                null, null, connectOptions, authCookie);
+                null, null, connectOptions, authCookie, null);
+    }
+
+    /**
+     * Constructs a new instance of this class and connects to the cloudant account with the
+     * specified credentials
+     *
+     * @param account        For cloudant.com, the cloudant account to connect to. For Cloudant
+     *                       local, the server URL
+     * @param interceptor Cookie interceptor with authentication
+     */
+    public CloudantClient(String account, CookieInterceptor interceptor) {
+        super();
+        Map<String, String> h = parseAccount(account);
+
+        doInit(h.get("scheme"),
+                h.get("hostname"),
+                new Integer(h.get("port")).intValue(),
+                h.get("username"),
+                h.get("password"),
+                null,
+                null, //authcookie
+                interceptor
+        );

```

Comment:
I don't think we should have any API exposure of the `CookieInterceptor` here. We should also leave the constructor argument changes for the other ticket.